### ISSUE TITLE
postConceptSetDefinition does not appear to work.

### ISSUE THREAD

**Issue by BradleySlavikAstellas** [OHDSI/ROhdsiWebApi#224]
Sample code?

The usual code to connect to ATLAS via ROhdsiWebApi

Retrieve a concept set definition, then try to use it to create a copy of it.

```
conceptSet ← getConceptSetDefinition(conceptSetId = 518, baseUrl = webapi_root)
View(conceptSet)
postConceptSetDefinition(name = “BS Test concept”, conceptSetDefinition = conceptSet, baseUrl = webapi_root)
```

Here is the error, which might help shed light on the issue

`Error in .request(url, method = “PUT”, config = config, handle = handle, :
Bad Request (HTTP 400).`

Is there a permission problem? Is our ATLAS instance misconfigured?
I have been using R programs to post cohort definitions and that code is working well.

Thanks for any suggestions on what I should do differently.

Bradley Slavik

P.S. I posted this on the OHDSI forum on June 28, but no one has responded to it.

**Comment by alondhe** [comment]
I faced this issue earlier, and posted a fix in the "develop" branch.

Can you try using it?

`devtools::install_github("OHDSI/ROhdsiWebApi", ref = "develop")`

**Comment by BradleySlavikAstellas** [comment]
Hello Ajit,

Not sure how to do that. Our RStudio is not allowed to connect to github. I don’t know how to download a different branch as a zip file. If I can figure that out, I can try to install new R version. I have to get others involved for any changes to the ATLAS side.

If this is not working, why hasn’t it been rolled into the main version? Surely someone wants to upload concept sets.

Bradley

**Comment by alondhe** [comment]
It looks like if you switch to the [develop branch](https://github.com/OHDSI/ROhdsiWebApi/tree/develop) and then Code -> Download

![image](https://user-images.githubusercontent.com/8037742/126674334-02fbedf2-d734-406d-877a-6f1fc9c40f16.png)

You get a zip file of the branch.

**Comment by BradleySlavikAstellas** [comment]
Hello Ajit,

Our RStudio had trouble installing packages so I finally got the develop branch installed to test today.
As you pointed out, it worked fine using the develop branch.

Some questions:


  1.  Why has this essential fix not been merged into the main library yet? It has been over a year.
  2.  Why was there no response on the forum question I posted in June? I had hoped that questions on the Forum would generate more activity. Does the OHDSI community primarily respond to bug reports?
  3.  Why isn’t the documentation synchronized with the code? If a function does not work, I expect the document to say “Unimplemented” until the main branch is working. I would never have written the code to use postConceptSetDefinition if the documentation had indicated that it was not working yet.

I look forward to being able to use this function.

Thank you very much for your assistance.

Sincerely,

Bradley Slavik

**Comment by alondhe** [comment]
@BradleySlavikAstellas:

_1.  Why has this essential fix not been merged into the main library yet? It has been over a year._

PRs need to be merged into develop and tested there before going into master. @gowthamrao -- can you comment on when the next release to master is expected?

_2.  Why was there no response on the forum question I posted in June? I had hoped that questions on the Forum would generate more activity. Does the OHDSI community primarily respond to bug reports?_

I'd say the Forums are for methods discussion, Git issues is the best place for targeted technical discussion.

_3.  Why isn’t the documentation synchronized with the code? If a function does not work, I expect the document to say “Unimplemented” until the main branch is working. I would never have written the code to use postConceptSetDefinition if the documentation had indicated that it was not working yet._

I'm not sure about the journey of this particular function, but I think largely this is the case. Perhaps the unit tests for it were lacking and therefore the bug was missed.

**Comment by BradleySlavikAstellas** [comment]
Nope. I don't know what happened here, but the code from the DEV branch that fixed this issue certainly did not make it into the 1.2.1 release. I had to use postConceptSetDefinition this week and I have been unable to get it to work. Same error. Also, I have been trying different DEV releases and have been unable to find the one with code that works. Are we certain that the relevant code is in release 1.2.1? Please re-open issue. I am happy to help by writing test cases. Just point me to code which works. Probably not relevant, but we are running ATLAS 2.7.8. Just in case that affects the issue.
